state/syncer/userAccountsSyncer.go

```diff
@@ -325,6 +333,29 @@ func (u *userAccountsSyncer) resetTimeoutHandlerWatchdog() {
 	u.timeoutHandler.ResetWatchdog()
 }
 
+// MissingDataTrieNodeFound is called whenever a missing data trie node is found.
+// This will trigger the sync process for the whole sub trie, starting from the given hash.
+func (u *userAccountsSyncer) MissingDataTrieNodeFound(hash []byte) {
+	u.mutex.Lock()
+	defer u.mutex.Unlock()
```

Comment:
this will prevent executing multiple sync processes at a time, suppose that a SC call will find 2 missing nodes that are unable to be synced in parallel, which improves performance

Reply:
Exactly, the same component is used in multiple places, so this will prevent any data races.

Reply:
add a TODO to think about this in the future?

Reply:
Also, for the in-processing trie syncer, I would opt for enabling the flag that checks on the disk if some of the trie data is present so we will minimize the p2p usage. We could think about this also

Reply:
Checked, and it should be safe to remove the mutex. Also, when `MissingDataTrieNodeFound` is called, the syncer will be created so that it will also check for nodes on disk..